lightly_studio/src/lightly_studio/api/routes/api/embeddings2d.py

```diff
@@ -35,23 +34,24 @@ def get_2d_embeddings(
     body: GetEmbeddings2DRequest | None = None,
 ) -> Response:
     """Return 2D embeddings serialized as an Arrow stream."""
-    # TODO(Malte, 09/2025): Support choosing the dataset via API parameter.
-    dataset = session.exec(select(DatasetTable).limit(1)).first()
-    if dataset is None:
-        raise HTTPException(status_code=404, detail="No dataset configured.")
+    dataset_id = (
+        body.filters.sample_filter.dataset_id
+        if body and body.filters and body.filters.sample_filter
+        else None
+    )
+    if dataset_id is None:
+        raise ValueError("Dataset ID must be provided in filters.")
```

Comment:
Changed HTTPExceptions to ValueErrors. Otherwise the error was swallowed without a print.